### ISSUE TITLE
Scratchpad show then floating toggle crashes sway.

### ISSUE THREAD

**Issue by Gomme-** [swaywm/sway#1468]
The following sequence of commands on a window makes sway to crach : 
move to scratchpad
scratchpad show
floating toggle.

Can someone confirm? 
(And question : what is the correct way to get a window out of scratchpad?)

**Comment by 4e554c4c** [comment]
Cannot confirm. This is the correct way to get a container out of the scratchpad.
Please attach a debug log and coredump so that we can debug your crash.

**Comment by Gomme-** [comment]
Here is the debug log:
[sway.log](https://github.com/swaywm/sway/files/1481602/sway.log)

**Comment by kasicka** [comment]
I have similar problem, sway usually crashes once per day or two. Initially it was crashing far more often, then I noticed FF57 consuming all the ram, disabling extensions helped a bit, but sway still crashes sometimes.
sway version is 0.15, wlc is 0.0.11
Log:
D: 11/15/17 09:30:20 - [ipc-server.c:249] Client 50 readable
D: 11/15/17 09:30:20 - [ipc-server.c:983] Added IPC reply to client 50 queue: [ { "id": 2, "name": "1", "rect": { "x": 0, "y": 23, "width": 1920, "height": 1057 }, "visible": false, "border": "none", "window_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "deco_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "geometry": { "x": 0, "y": 0, "width": 0, "height": 0 }, "percent": null, "window": -1, "urgent": false, "current_border_width": 0, "num": 1, "output": "eDP-1", "type": "workspace", "layout": "splith", "focused": false }, { "id": 6, "name": "2", "rect": { "x": 0, "y": 23, "width": 1920, "height": 1057 }, "visible": false, "border": "none", "window_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "deco_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "geometry": { "x": 0, "y": 0, "width": 0, "height": 0 }, "percent": null, "window": -1, "urgent": false, "current_border_width": 0, "num": 2, "output": "eDP-1", "type": "workspace", "layout": "splith", "focused": false }, { "id": 55, "name": "6", "rect": { "x": 0, "y": 23, "width": 1920, "height": 1057 }, "visible": true, "border": "none", "window_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "deco_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "geometry": { "x": 0, "y": 0, "width": 0, "height": 0 }, "percent": null, "window": -1, "urgent": false, "current_border_width": 0, "num": 6, "output": "eDP-1", "type": "workspace", "layout": "splith", "focused": true }, { "id": 11, "name": "7", "rect": { "x": 0, "y": 23, "width": 1920, "height": 1057 }, "visible": false, "border": "none", "window_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "deco_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "geometry": { "x": 0, "y": 0, "width": 0, "height": 0 }, "percent": null, "window": -1, "urgent": false, "current_border_width": 0, "num": 7, "output": "eDP-1", "type": "workspace", "layout": "splith", "focused": false }, { "id": 40, "name": "8", "rect": { "x": 0, "y": 23, "width": 1920, "height": 1057 }, "visible": false, "border": "none", "window_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "deco_rect": { "x": 0, "y": 0, "width": 0, "height": 0 }, "geometry": { "x": 0, "y": 0, "width": 0, "height": 0 }, "percent": null, "window": -1, "urgent": false, "current_border_width": 0, "num": 8, "output": "eDP-1", "type": "workspace", "layout": "splith", "focused": false } ]
D: 11/15/17 09:30:20 - [ipc-server.c:315] Client 50 writable
I: 11/15/17 09:30:20 - [commands.c:425] Handling command 'scratchpad show'
D: 11/15/17 09:30:20 - [commands.c:344] find_handler(scratchpad) 0
I: 11/15/17 09:30:20 - [commands.c:469] Running on container '6'
D: 11/15/17 09:30:20 - [layout.c:140] Adding 0x27f75d0 (4, 960.000000x792.750000) to 0x37a2ef0 (2, 1920.000000x1057.000000)
D: 11/15/17 09:30:20 - [ipc-server.c:1080] Sending window::floating event
E: 11/15/17 09:30:21 - [bar.c:275] failed to dispatch wl: 104
E: 11/15/17 09:30:21 - [ipc-client.c:50] Unable to receive IPC response
D: 11/15/17 09:30:21 - [main.c:52] [wlc] Parent exit (2025)
D: 11/15/17 09:30:21 - [main.c:52] [wlc] Restoring vt 1 (0x4) (fd 0)
Gdk-Message: /usr/lib64/firefox/firefox: Fatal IO error 11 (Resource temporarily unavailable) on X server :0.

Pidgin: Fatal IO error 11 (Resource temporarily unavailable) on X server :0.
Gdk-Message: firefox: Fatal IO error 0 (Success) on X server :0.

**Comment by SirCmpwn** [comment]
Can I get a backtrace?

**Comment by Gomme-** [comment]
I am sorry I don't know what it is. How can I get that?

**Comment by SirCmpwn** [comment]
Ask your distro for help obtaining a coredump and feed it into gdb to get a stack trace.

**Comment by icetan** [comment]
I have the same problem. When after starting `sway` and following the instructions I get a crash every time.

I'm on Arch Linux

```
$ uname -a
Linux pix 4.13.8-6-ph #1 SMP Mon Oct 23 14:02:48 CEST 2017 x86_64 GNU/Linux
$ pacman -Q sway wlc
sway 0.15.0-1
wlc 0.0.11-1
```

**Comment by Gomme-** [comment]
Yes it seems to happen just after starting sway with one window. 
By the way, sorry for not providing a backtrace yet. I am quite busy now and still don't know how to get that. I might have time to read the archwiki carefully and provide you the backtrace next week.

**Comment by Timidger** [comment]
@Gomme-

These instructions should work for arch:

1) run sway, make it crash
2) run `coredumpctl gdb` (make sure you have gdb installed)
3) type `backtrace` into the prompt, past output in this issue tracker

**Comment by nickbatsaras** [comment]
I have a similar issue. Following the exact steps below, crashes sway everytime:
1. Move a window to scratchpad
2. Move another window to scratchpad
3. Scratchpad show
4. Floating toggle
5. Scratchpad show
6. Scratchpad show

I am using sway version 0.15.0 on Arch.

**Comment by bzub** [comment]
I experience this, also on Arch.

@Timidger @SirCmpwn When sway crashes it doesn't appear to generate a coredump, according to `coredumpctl list`. From time to time manipulating firefox windows seems to crash sway as well, and does generate a coredump. So I'm posting one of those with backtrace in case it's related to this issue.

[backtrace.txt](https://github.com/swaywm/sway/files/1613019/backtrace.txt)

**Comment by Timidger** [comment]
That doesn't help much. It doesn't have any debug symbols for one (unsurprising) but also because even if a client crashes Sway should handle it gracefully. If you can figure out why Sway crashes (and get a coredump of that, with debug symbols, if it is actually segfaulting) it would make it easier to debug.

**Comment by bzub** [comment]
This issue seems to have gone away since my last comment. I upgraded sway to 0.15.1, so @Gomme- @icetan you might want to try that version.

**Comment by Gomme-** [comment]
Sadly, I still have the problem with 0.15.1.

**Comment by HarmlessBison0** [comment]
I experienced this as well, exactly as described by @nickbatsaras.
Running Fedora 27 with Sway 0.15.1 x86_64

**Comment by naibaf0** [comment]
I can confirm the same behaviour on Arch with `0.15.1-2.`
```
I: 03/20/18 00:15:22 - [main.c:69] Using output 0 of 1
Warning: 'icon_folders' is deprecated, please use 'icon_path' instead.
Cannot acquire 'org.freedesktop.Notifications': Name is acquired by PID '1175'.
E: 03/20/18 00:15:31 - [debug_log.c:88] swayc_log:swayc_log: no container ...
E: 03/20/18 00:15:31 - Gdk-Message: Error reading events from display: Broken pipe
[ipc-client.c:50] Unable to receive IPC response

** (xss-lock:6546): CRITICAL **: X connection lost; exiting.
```
here is what I get in the log of sway, no coredump/backtrace seems to happen for me either.

However this only happened for me when running sway in parallel to a normal xsession with i3. If I started sway and nothing else the problem didn't occur.

**Comment by SirCmpwn** [comment]
I'm closing issues that are resolved by the yet-unreleased sway 1.0.
